### PR TITLE
[MGDOBR-899] Implement E2E test to allow a user to update Action Para…

### PR DESCRIPTION
**Pull request by sthundat**
…meters on a Sink Processor

[MGDOBR-899](https://issues.redhat.com/browse/MGDOBR-899)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

### CODE REVIEW

File: integration-tests/src/test/resources/features/action-slack.feature [hunk 1]
```diff
@@ -40,3 +40,70 @@ Feature: Slack Action tests
...
+      "name": "slackProcessor",
+      "action": {
+        "type": "slack_sink_0.1",
+        "parameters": {
+            "slack_channel": "${slack.channel.general.name}",
+            "slack_webhook_url": "${slack.channel.mc.webhook.url}"
+          }
+      }
+    }
+    """
```
Comment (jomarko) [inline integration-tests/src/test/resources/features/action-slack.feature:90]: Could you please clarify our intention of sending cloud event to the 'mybridge' two times? I understand that both should be forwarded to different slack webhook urls, however I as we check just one slack channel, the later one, to me sending the first cloud event seems like not needed, rather slowing down the test?

To put the question, can't we rewrite the test to send just one cloud event and speed up the test in that way? 

Just asking, maybe there is some aspect I do not see.

**Review by jomarko — COMMENTED** [review]
Thank you for a PR @sthundat

File: integration-tests/src/test/resources/features/action-slack.feature [hunk 2]
```diff
@@ -40,3 +40,70 @@ Feature: Slack Action tests
...
+      "id": "test-id",
+      "time": "2019-11-18T15:13:39.4589254Z",
+      "subject": "blobServices/default/containers/{storage-container}/blobs/{new-file}",
+      "dataschema": "#",
+      "data": {
+        "name": "Hello world!"
+        }
+    }
+    """
+
```
Comment (sutaakar) [inline integration-tests/src/test/resources/features/action-slack.feature:77]: @sthundat here you should check that the message arrived to Slack `general` channel

File: integration-tests/src/test/resources/features/action-slack.feature [hunk 3]
```diff
@@ -40,3 +40,70 @@ Feature: Slack Action tests
...
+    And update existing Processor with modified webhook url to the Bridge "mybridge" with body:
+    """
+    {
+      "name": "slackProcessor",
+      "action": {
+        "type": "slack_sink_0.1",
+        "parameters": {
+            "slack_channel": "${slack.channel.general.name}",
```
Comment (sutaakar) [inline integration-tests/src/test/resources/features/action-slack.feature:85]:
```suggestion
            "slack_channel": "${slack.channel.mc.name}",
```

File: integration-tests/src/test/resources/features/action-slack.feature (hunk 1 above)
Comment (sutaakar) [reply integration-tests/src/test/resources/features/action-slack.feature:90]: IMHO we should check both channels

Comment (sthundat) [reply integration-tests/src/test/resources/features/action-slack.feature]: Both the channels are checking now

File: integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/ProcessorSteps.java [hunk 4]
```diff
@@ -239,4 +239,30 @@ private Source getProcessorSource(String processorName, String testBridgeName) {
         return ProcessorResource.getProcessor(context.getManagerToken(),
                 bridgeContext.getId(), processorId).getSource();
     }
+
+    @And("update existing Processor with modified webhook url to the Bridge {string} with body:")
```
[the remark this answers is not in the document]
Comment (sthundat) [reply integration-tests/src/test/java/com/redhat/service/smartevents/integration/tests/steps/ProcessorSteps.java]: I thought `updateProcessorOfTheBridgeWithBody` is removing the existing processor.
Removed the code.

**Review by sutaakar — APPROVED** [review]

**Review by Srihari1192 — APPROVED** [review]
LGTM

**Review by jomarko — APPROVED** [review]
Thank you @sthundat

**Pull request merged**